File: samples/android-core-sample-java/app/src/main/java/com/example/androidcoresamplejava/MainActivity.java
package com.example.androidcoresamplejava;

import android.util.Log;
import android.view.View;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.ProgressBar;
import io.dyte.core.DyteMeetingBuilder;
import io.dyte.core.DyteMobileClient;
import io.dyte.core.listeners.DyteMeetingRoomEventsListener;
import io.dyte.core.models.ActiveTabType;
import io.dyte.core.models.DyteMeetingInfoV2;

public class MainActivity extends AppCompatActivity {
  private String TAG = "MainActivity";

  private Button btnInit, btnJoin, btnLeave;
  private ProgressBar pbLoader;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    btnInit = findViewById(R.id.btnInitMeeting);
    btnJoin = findViewById(R.id.btnJoinMeeting);
    btnLeave = findViewById(R.id.btnLeaveMeeting);
    pbLoader = findViewById(R.id.pbLoader);

    btnJoin.setVisibility(View.GONE);
    btnLeave.setVisibility(View.GONE);

    DyteMobileClient mobileClient = DyteMeetingBuilder.INSTANCE.build(this);
    DyteMeetingRoomEventsListener meetingEventListener = new DyteMeetingRoomEventsListener() {
      @Override public void onReconnectingToMeetingRoom() {

      }

      @Override public void onReconnectedToMeetingRoom() {

      }

      @Override public void onMeetingRoomReconnectionFailed() {

      }

      @Override public void onMeetingRoomDisconnected() {

      }

      @Override public void onMeetingRoomConnectionFailed() {

      }

      @Override public void onMeetingRoomLeaveStarted() {
        Log.v(TAG, "onMeetingRoomLeaveStarted");
      }

      @Override public void onMeetingRoomLeaveCompleted() {
        Log.v(TAG, "onMeetingRoomLeaveCompleted");
      }

      @Override public void onMeetingRoomJoinStarted() {
        Log.v(TAG, "onMeetingRoomJoinStarted");
        btnJoin.setEnabled(false);
        pbLoader.setVisibility(View.VISIBLE);
      }

      @Override public void onMeetingRoomJoinFailed(@NonNull Exception e) {
        Log.v(TAG, "onMeetingRoomJoinFailed");
      }

      @Override public void onMeetingRoomJoinCompleted() {
        Log.v(TAG, "onMeetingRoomJoinCompleted");
        pbLoader.setVisibility(View.GONE);
        btnJoin.setVisibility(View.GONE);
        btnLeave.setVisibility(View.VISIBLE);
      }

      @Override public void onMeetingInitStarted() {
        Log.v(TAG, "onMeetingInitStarted");
        btnInit.setEnabled(false);
        pbLoader.setVisibility(View.VISIBLE);
      }

      @Override public void onMeetingInitFailed(@NonNull Exception e) {
        Log.v(TAG, "onMeetingInitFailed " + e.getMessage());
      }

      @Override public void onMeetingInitCompleted() {
        Log.v(TAG, "onMeetingInitCompleted");
        pbLoader.setVisibility(View.GONE);
        btnInit.setVisibility(View.GONE);
        btnJoin.setVisibility(View.VISIBLE);
      }

      @Override public void onDisconnectedFromMeetingRoom() {

      }

      @Override public void onConnectingToMeetingRoom() {

      }

      @Override public void onConnectedToMeetingRoom() {

      }

      @Override
      public void onActiveTabUpdate(@NonNull String s, @NonNull ActiveTabType activeTabType) {

      }
    };

    mobileClient.addMeetingRoomEventsListener(meetingEventListener);

    btnInit.setOnClickListener(view -> {
      Log.v(TAG, "doing init meeting");
      DyteMeetingInfoV2 dyteMeetingInfoV2 = new DyteMeetingInfoV2(
          Constants.AUTH_TOKEN,
          true, true, Constants.BASE_URL);
      mobileClient.init(dyteMeetingInfoV2);
    });

    btnJoin.setOnClickListener(view -> {
      Log.v(TAG, "doing join meeting");
      mobileClient.joinRoom();
    });
  }
}

File: samples/android-core-sample-kotlin/app/src/main/java/com/example/androidcoresamplekotlin/MainActivity.kt
package com.example.androidcoresamplekotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import io.dyte.core.DyteAndroidClientBuilder
import io.dyte.core.listeners.DyteMeetingRoomEventsListener
import io.dyte.core.listeners.DyteParticipantEventsListener
import io.dyte.core.models.DyteJoinedMeetingParticipant
import io.dyte.core.models.DyteMeetingInfoV2

private const val TAG = "DyteMeeting"

class MainActivity : AppCompatActivity() {
  private lateinit var btnInitMeeting: Button
  private lateinit var btnJoinMeeting: Button
  private lateinit var btnLeaveMeeting: Button
  private lateinit var pbLoader: ProgressBar

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    btnInitMeeting = findViewById(R.id.btnInitMeeting)
    btnJoinMeeting = findViewById(R.id.btnJoinMeeting)
    btnLeaveMeeting = findViewById(R.id.btnLeaveMeeting)
    pbLoader = findViewById(R.id.pbLoader)

    btnInitMeeting.visibility = View.VISIBLE
    btnJoinMeeting.visibility = View.GONE
    btnLeaveMeeting.visibility = View.GONE
    pbLoader.visibility = View.GONE

    val meetingEventListener = object : DyteMeetingRoomEventsListener {
      override fun onMeetingInitStarted() {
        Log.v(TAG, "onMeetingInitStarted")
        btnInitMeeting.isEnabled = false
        pbLoader.visibility = View.VISIBLE
      }

      override fun onMeetingInitCompleted() {
        Log.v(TAG, "onMeetingInitCompleted")
        pbLoader.visibility = View.GONE
        btnInitMeeting.isEnabled = true
        btnInitMeeting.visibility = View.GONE
        btnJoinMeeting.visibility = View.VISIBLE
      }

      override fun onMeetingInitFailed(exception: Exception) {
        Log.v(TAG, "onMeetingInitFailed ${exception.message}")
      }

      override fun onMeetingRoomJoinStarted() {
        Log.v(TAG, "onMeetingRoomJoinStarted")
        btnJoinMeeting.isEnabled = false
        pbLoader.visibility = View.VISIBLE
      }

      override fun onMeetingRoomJoinCompleted() {
        Log.v(TAG, "onMeetingRoomJoinCompleted")
        pbLoader.visibility = View.GONE
        btnInitMeeting.isEnabled = true
        btnJoinMeeting.visibility = View.GONE
        btnLeaveMeeting.visibility = View.VISIBLE
      }

      override fun onMeetingRoomJoinFailed(exception: Exception) {
        Log.v(TAG, "onMeetingRoomJoinFailed ${exception.message}")
      }
    }

    val participantEventsListener = object : DyteParticipantEventsListener {
      override fun onActiveParticipantsChanged(active: List<DyteJoinedMeetingParticipant>) {
        super.onActiveParticipantsChanged(active)
        Log.v(TAG, "onActiveParticipantsChanged ${active.size}")
      }

      override fun onParticipantJoin(participant: DyteJoinedMeetingParticipant) {
        super.onParticipantJoin(participant)
        Log.v(TAG, "onParticipantJoin ${participant.name}")
      }

      override fun onParticipantLeave(participant: DyteJoinedMeetingParticipant) {
        super.onParticipantLeave(participant)
        Log.v(TAG, "onParticipantJoin ${participant.name}")
      }
    }

    val dyteMeetingInfo =
      DyteMeetingInfoV2(authToken = AUTH_TOKEN, enableAudio = false, enableVideo = false)
    val meeting = DyteAndroidClientBuilder.build(this)
    meeting.addMeetingRoomEventsListener(meetingEventListener)
    meeting.addParticipantEventsListener(participantEventsListener)

    val chat = meeting.chat
    val roomParticipants = meeting.participants
    val polls = meeting.polls
    val localUser = meeting.localUser

    btnInitMeeting.setOnClickListener {
      meeting.init(dyteMeetingInfo)
    }

    btnJoinMeeting.setOnClickListener {
      meeting.joinRoom()
    }

    btnLeaveMeeting.setOnClickListener {
      meeting.leaveRoom()
    }
  }
}